### PR TITLE
[14.0][MIG] osi_bank_reconciled_report

### PR DESCRIPTION
**Pull request by Freni-OSI**

### CODE REVIEW

**Comment by patrickrwilson** [comment]
@Freni-OSI I understand this might still be in progress, I tried installing and got the following error:
  File "C:\Odoo\Odoo 14.0e\server\odoo\addons\osi_bank_reconciled_report\models\account_bank_reconciled_report.py", line 111, in _get_bank_rec_report_data
    self.env.context["company_ids"]
Exception

The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 639, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 315, in _handle_exception
    raise exception.with_traceback(None) from new_cause
KeyError: 'company_ids'

Please let me know when it's ready for my review.

**Comment by Freni-OSI** [comment]
> @Freni-OSI I understand this might still be in progress, I tried installing and got the following error:
> File "C:\Odoo\Odoo 14.0e\server\odoo\addons\osi_bank_reconciled_report\models\account_bank_reconciled_report.py", line 111, in _get_bank_rec_report_data
> self.env.context["company_ids"]
> Exception
> 
> The above exception was the direct cause of the following exception:
> Traceback (most recent call last):
> File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 639, in _handle_exception
> return super(JsonRequest, self)._handle_exception(exception)
> File "C:\Odoo\Odoo 14.0e\server\odoo\http.py", line 315, in _handle_exception
> raise exception.with_traceback(None) from new_cause
> KeyError: 'company_ids'
> 
> Please let me know when it's ready for my review.

File: osi_bank_reconciled_report/models/account_bank_reconciled_report.py [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+        lines.append(self._add_title_line(options, _("Operations to Process"), level=1))
+
+        if report_data.get("not_reconciled_st_positive") or report_data.get(
+            "not_reconciled_st_negative"
+        ):
+            lines.append(
+                self._add_title_line(
+                    options,
+                    _(
+                        "Uneconciled Bank \
```
Comment (patrickrwilson) [inline osi_bank_reconciled_report/models/account_bank_reconciled_report.py]: Misspelling, please change to Uneconciled to Unreconciled. v12 has an active PR to fix this which is why it wasn't part of the code here.

**Review by patrickrwilson — CHANGES_REQUESTED** [review]
Fix misspelling then should be good.

File: osi_bank_reconciled_report/models/account_bank_reconciled_report.py (hunk 1 above)
Comment (Freni-OSI) [reply osi_bank_reconciled_report/models/account_bank_reconciled_report.py]: Fixed. Thanks

**Review by patrickrwilson — APPROVED** [review]
Tested, LGTM

**Review by osi-scampbell — APPROVED** [review]

**Review by smangukiya — APPROVED** [review]

**Pull request merged**